File: driving-school-client/src/app/shared/model/Student.ts
import {IUser, User} from './User';

export class Student implements IStudent{
  id: string;
  firstName: string;
  lastName: string;
  status: string;
  hours: number;
  address: string;
  email: string;
  phone: number;
  createDate: string;
  updateDate: string;
  user: User;
  categoryList?: string[];
}
export interface IStudent {
  id: string;
  firstName: string;
  lastName: string;
  status: string;
  hours: number;
  address: string;
  email: string;
  phone: number;
  createDate: string;
  updateDate: string;
  user: IUser;
  categoryList?: string[];
}

File: driving-school-api/src/main/java/driving/school/model/Student.java
package driving.school.model;

import driving.school.model.user.User;
import lombok.*;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.ResultCheckStyle;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

@EqualsAndHashCode(callSuper = true)
@Data
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Entity
@SQLDelete(sql = "update student set deleted = true where id = ?", check = ResultCheckStyle.COUNT)
@Where(clause = "deleted <> true")
public class Student extends BaseEntity {
    private String firstName;
    private String lastName;
    private String address;
    private String email;
    private Integer hours;
    private String phone;

    @Enumerated(EnumType.STRING)
    private StudentStatus status;

    @OneToMany(mappedBy = "student")
    private List<RideDate> rideDateList;

    @Enumerated(EnumType.STRING)
    @ElementCollection(fetch = FetchType.EAGER)
    private Set<Category> categoryList;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    public Student(Long id) {
        super(id);
    }
}
